initrd/etc/functions

```diff
@@ -19,18 +19,28 @@ mask_param() {
 # N=0 is the name of the command to be executed, N=1 is its first parameter,
 # etc.
 DO_WITH_DEBUG() {
-	if [ "$1" == "--mask-position" ]; then
-		mask_position="$2"
+	local exit_status
+	local cmd_output
+	DEBUG "PATH: $PATH"
+	local cmd=("$@")
+	if [[ "$1" == "--mask-position" ]]; then
+		local mask_position="$2"
 		shift
 		shift
-		DEBUG_ARGS=("$@")
-
-		DEBUG_ARGS[$mask_position]="$(mask_param "${DEBUG_ARGS[$mask_position]}")"
-		DEBUG "${DEBUG_ARGS[@]}"
-	else
-		DEBUG "$@"
+		cmd=("$@")
+		cmd[$mask_position]="$(mask_param "${cmd[$mask_position]}")"
+	fi
+	if [[ ${#cmd[@]} -eq 1 ]]; then
+		# If there's only one argument, try to split it into multiple arguments
+		read -a cmd <<< "${cmd[0]}"
 	fi
```

Comment:
I don't think this "automatic split" is a good idea.  The caller should know whether the argument needs to be split or not.

If you try to do this automatically, it will never be clear whether callers intend for an argument to be split or not.  We'll start introducing subtle errors when (say) something usually doesn't need to be split, but occasionally ends up with an embedded argument that does (and we don't realize it in refactoring), etc.

Keep it clear whether the argument should be split or not.  I think the only usage of this right now is the `eval` in kexec that was removed, just put it back.